packages/x-charts/src/ChartsLegend/ContinuousColorLegend.tsx

```diff
@@ -261,7 +261,7 @@ const ContinuousColorLegend = consumeThemeProps(
             rotate={rotateGradient}
             reverse={reverse}
             thickness={thickness}
-            gradientId={gradientId ?? generateGradientId(axisItem.id, axisDirection!)}
+            gradientId={gradientId ?? generateGradientId(axisItem.id)}
```

Comment:
@alexfauquette I've removed `direction=(x,y,z)` from usage. It didn't look useful, as `axisId` should technically be unique by itself.

Reply:
> The `axisId` should technically be unique by itself

It's a debate I got when creating the lib. Should axis id be unique in general or unique per direction.

For now, we only enforce the "unique per direction" (If not respected users will get a warning). If you remove the axis-driection in the id, users could get weird resultsaxis-direction without any error in the console.

Duplicated ids are a bit tricky to identify as bug. That's why I preferred to be conservative on it. But that's up to debate

Reply:
Feels a bit useless, `id` is generally expected to be unique. Although you could argue that axis direction is a different domain 🙃.

Should we hold onto this while we wait for the plugins to merge? then I can try to reason with how we currently handle the warnings ad make a decision 😆

Reply:
> Although you could argue that axis direction is a different domain 🙃

That's exactly my argument :p The main usecase I've in mind is a scatter plot where you can any data grid column versus any other column.

Then the chart has the same set of x and y-axes to chose from. And so either we or the developer have to be careful about avoiding id colision.

By order of preferences, we can

1. Enforce id unicity
2. Force adding x/y prefix
3. Hope user have unique id

So I agree about waiting for the plugins to be merged and verify if we can enforce id unicity